chain/sync.go

```diff
@@ -166,11 +171,31 @@ func NewSyncer(sm *stmgr.StateManager, bsync *blocksync.BlockSync, connmgr connm
 }
 
 func (syncer *Syncer) Start() {
+	syncer.ticker = build.Clock.Ticker(time.Duration(build.BlockDelaySecs) * time.Second)
+	syncer.tickerCtx, syncer.tickerCtxCancel = context.WithCancel(context.Background())
 	syncer.syncmgr.Start()
+
+	genesisTime := time.Unix(int64(syncer.Genesis.MinTimestamp()), 0)
+
+	go func() {
+		for {
+			select {
+			case <-syncer.ticker.C:
+				sinceGenesis := build.Clock.Now().Sub(genesisTime)
+				expectedHeight := int64(sinceGenesis.Seconds()) / int64(build.BlockDelaySecs)
+
+				stats.Record(syncer.tickerCtx, metrics.ChainNodeHeightExpected.M(int64(expectedHeight)))
+			case <-syncer.tickerCtx.Done():
+				return
+			}
+		}
+	}()
```

Comment:
I'd put this in a separate function, then just call as `go syncer.runMetricsTicker(tickerCtx)`